app/domain/authentication/authn_gcp/validate_resource_restrictions.rb

```diff
@@ -0,0 +1,53 @@
+module Authentication
+  module AuthnGcp
+
+    # This class is responsible of validating resource restrictions that configured on a Conjur host or user
+    # against google JWT token.
+    ValidateResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions:                ExtractResourceRestrictions.new,
+        validate_resource_restrictions_configuration: ValidateResourceRestrictionsConfiguration.new,
+        validate_resource_restrictions_match_jwt:     ValidateResourceRestrictionsMatchJWT.new,
+        logger:                                       Rails.logger
+      },
+      inputs:       [:authenticator_input]
+    ) do
+
+      extend Forwardable
+      def_delegators :@authenticator_input, :account, :username, :credentials
+
+      def call
+        @logger.debug(LogMessages::Authentication::ValidatingResourceRestrictions.new)
+        extract_resource_restrictions
+        validate_resource_restrictions_configuration
+        validate_resource_restrictions_match_jwt
+        @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictions.new)
+      end
+
+      private
+
+      def extract_resource_restrictions
+        @resource_restrictions ||= @extract_resource_restrictions.call(
+          account:           account,
+          username:          username,
+          extraction_prefix: AUTHN_PREFIX
+        )
+      end
```

Comment:
when you use memoization you need the memoized object name to match the method name. what you could do is:
```
      def extract_resource_restrictions
        resource_restrictions
      end

      def resource_restrictions
        @resource_restrictions ||= @extract_resource_restrictions.call(
          account:           account,
          username:          username,
          extraction_prefix: AUTHN_PREFIX
        )
      end
```